hips/tiles/tests/test_tile.py

```diff
@@ -1,46 +1,87 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
+import pytest
 import numpy as np
 from astropy.tests.helper import remote_data
 from numpy.testing import assert_allclose
 from numpy.testing import assert_equal
 from ..tile import HipsTile, HipsTileMeta
-
+from ...utils.testing import get_hips_extra_file, requires_hips_extra
 
 class TestHipsTile:
     @remote_data
-    def test_fetch_read_write_fits(self, tmpdir):
-        meta = HipsTileMeta(order=6, ipix=30889, file_format='fits')
-        url = 'http://alasky.unistra.fr/2MASS/H/Norder6/Dir30000/Npix30889.fits'
+    def test_fetch_fits(self, tmpdir):
+        meta = HipsTileMeta(order=3, ipix=463, file_format='fits')
+        url = 'http://alasky.unistra.fr/DSS/DSS2Merged/Norder3/Dir0/Npix463.fits'
         tile = HipsTile.fetch(meta, url)
 
         assert tile.data.shape == (512, 512)
-        assert_equal(tile.data[510][5:7], [1, 0])
+        assert_equal(tile.data[510][5:7], [3047, 3257])
 
-        filename = str(tmpdir / 'Npix30889.fits')
+        filename = str(tmpdir / 'Npix463.fits')
         tile.write(filename)
-        tile2 = HipsTile.read(meta, filename=filename)
+        tile2 = HipsTile.read(meta, full_path=filename)
 
         assert tile == tile2
 
     @remote_data
-    def test_fetch_read_write_jpg(self, tmpdir):
-        meta = HipsTileMeta(order=6, ipix=30889, file_format='jpg')
-        url = 'http://alasky.unistra.fr/2MASS/H/Norder6/Dir30000/Npix30889.jpg'
+    def test_fetch_jpg(self, tmpdir):
+        meta = HipsTileMeta(order=3, ipix=463, file_format='jpg')
+        url = 'http://alasky.unistra.fr/DSS/DSS2Merged/Norder3/Dir0/Npix463.jpg'
         tile = HipsTile.fetch(meta, url)
 
         assert tile.data.shape == (512, 512, 3)
-        assert_equal(tile.data[510][5:7], [[0, 0, 0], [1, 1, 1]])
+        assert_equal(tile.data[510][5:7], [[10, 10, 10], [11, 11, 11]])
 
-        filename = str(tmpdir / 'Npix30889.jpg')
+        filename = str(tmpdir / 'Npix463.jpg')
         tile.write(filename)
-        tile2 = HipsTile.read(meta, filename=filename)
+        tile2 = HipsTile.read(meta, full_path=filename)
 
         # The following assert fails, because on JPEG write / read
         # the data is different (for unknown reasons).
         # TODO: Figure out what's wrong here and fix!
-        # print(tile.data.sum())
+        # print(tile1.data.sum())
         # print(tile2.data.sum())
-        # print((tile == tile2).all())
+        # print((tile1 == tile2).all())
+        # assert tile1 == tile2
+
+    @remote_data
+    def test_fetch_png(self, tmpdir):
```

Comment:
Suggest to change all fetch tests like this:

* Use a tile from hips-extra
* Fetch it from the raw URL like e.g. https://github.com/hipspy/hips-extra/raw/master/datasets/samples/2MASS6XH/Norder6/Dir0/Npix1517.png
* Also read it via Tile.read
* Assert tile1 == tile2
* Don't do any extra asserts or tile writing (instead do the assertions on shape or pixel values in the other, non-fetch tests below)